x/programs/rust/wasmlanche_sdk/src/host/state.rs

```diff
@@ -38,3 +41,13 @@ pub(crate) unsafe fn get_bytes(caller: &Program, key: &Key) -> Result<i64, State
     let key = to_host_ptr(key)?;
     Ok(unsafe { _get(caller, key) })
 }
+
+/// Deletes the bytes at key ptr from the host storage
+pub(crate) unsafe fn delete_bytes(caller: &Program, key: &Key) -> Result<(), StateError> {
+    let caller = to_host_ptr(caller.id())?;
+    let key = to_host_ptr(key)?;
+    match unsafe { _delete(caller, key) } {
+        0 => Ok(()),
+        _ => Err(StateError::Delete),
```

Comment:
🤔, is this the only thing that can go wrong here? What if the key doesn't exist vs a read-error etc?

Reply:
Following errors can occur golang side:
1)Fail to access caller memory
2)Fail to access/read program & key pointer
3)Fail to Remove from the State

Actually, any returned error by the extern golang function will result in a panic.
The only error I could handle is the third scenario returning `-1`, just like the `get_bytes`. But I cannot add more details than `StateError::Delete`
-> Going to handle this scenario.
There is the same problem with the `put_bytes` method returning `StateError::Write`. It will never happens.

https://github.com/ava-labs/hypersdk/issues/666 looks like this issue is talking about error enhancement.